stl/inc/algorithm

```diff
@@ -10007,13 +9943,79 @@ namespace ranges {
         _NODISCARD constexpr minmax_element_result<borrowed_iterator_t<_Rng>> operator()(
             _Rng&& _Range, _Pr _Pred = {}, _Pj _Proj = {}) const {
             auto _First   = _RANGES begin(_Range);
-            auto _UResult = _RANGES _Minmax_element_unchecked(
-                _Unwrap_range_iter<_Rng>(_STD move(_First)), _Uend(_Range), _Pass_fn(_Pred), _Pass_fn(_Proj));
-            _Seek_wrapped(_First, _STD move(_UResult.min));
+            auto _UResult = _Minmax_element_fwd_unchecked(_RANGES _Unwrap_range_iter<_Rng>(_STD move(_First)),
+                _RANGES _Uend(_Range), _STD _Pass_fn(_Pred), _STD _Pass_fn(_Proj));
+            _STD _Seek_wrapped(_First, _STD move(_UResult.min));
             auto _Second = _First;
-            _Seek_wrapped(_Second, _STD move(_UResult.max));
+            _STD _Seek_wrapped(_Second, _STD move(_UResult.max));
             return {_STD move(_First), _STD move(_Second)};
         }
+
+    private:
+        template <class _It, class _Se, class _Pr, class _Pj>
+        _NODISCARD static constexpr min_max_result<_It> _Minmax_element_fwd_unchecked(
+            _It _First, const _Se _Last, _Pr _Pred, _Pj _Proj) {

```

Comment:
libc++ has tests which assert that `_Pred` and `_Proj` are not copied.
https://github.com/llvm/llvm-project/blob/29c3a2c6dbce4b8232ea21944abc18a3e165d26a/libcxx/test/libcxx/algorithms/ranges_robust_against_copying_comparators.pass.cpp
https://github.com/llvm/llvm-project/blob/29c3a2c6dbce4b8232ea21944abc18a3e165d26a/libcxx/test/libcxx/algorithms/ranges_robust_against_copying_projections.pass.cpp

These tests are in `libcxx` test suite and not in `std` so probably it's not required by C++ standard.
But I decided to inform you.

Reply:
Note that `_Pass_fn` should avoid copying.

Reply:
Yes, I'm aware libc++ provides this extension. Our `_Pass_fn` copies function objects no larger than `void*`, and wraps larger function objects with a `reference_wrapper`-like type. This was the best compromise Billy found for performance: the optimizer really hates double indirections (references or pointers to references or pointers). People who _really_ need to avoid copies can always pass `std::ref(my_function)` if they like. (Aside: we apparently forgot to `if constexpr` the two overloads of `_Pass_fn` together. I'll make a PR.)